### DISCUSSION TITLE
Block password setting for OIDC user

### DISCUSSION THREAD

**Discussion by Sytten** [ory/kratos#3278] [Q&A]
Hi! I was wondering what would be the best way to prevent users registered with OIDC from setting a password via recovery (I am guessing they cant do it in the settings since  they dont know the current password). I was thinking maybe a before hook that stops the flow?

The idea here is that we are building an organization concept and I want the org admin to be able to deny user login with passwords (force login with social) for SSO reason. It's not a full on SSO but usually good enough for startups.

**Comment by aeneasr** [comment]
That's a good use case, but I don't think it's currently possible to implement this. Maybe though it is enough to do this in your frontend? So hide the password field if a user has SSO? The user could still set a password if they "hack" the API, but the impact of that is probably negligible :)

**Reply by Sytten** [reply]
Clearly not since my customers are literal hackers/pentesters 😂
